.github/workflows/testsuites.yml

```diff
@@ -0,0 +1,141 @@
+# SPDX-FileCopyrightText: 2024 Rivos Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: linux-riscv-ci-testsuites
+
+defaults:
+  run:
+    shell: bash -leo pipefail {0}
+
+on: pull_request
+
+concurrency:
+  group: ci-test-${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  build-kernel:
+    if:  ${{ endsWith(github.head_ref, '_test') }}
+    runs-on: self-hosted
+    timeout-minutes: 50400 # 35 days
+    container:
+      image: ghcr.io/linux-riscv/linaro-tuxrun-dispatcher-riscv64:latest
+      volumes:
+        - /home/github/ccache:/build/ccache
+        - /home/github/gitref:/build/gitref
+      options: --device /dev/kvm
+    steps:
+      - name: Configure git
+        run: |
+          git config --global --add safe.directory '*'
+      - name: Checkout git
+        run: |
+          mkdir -p /build/my-linux
+          cd /build/my-linux
+          git clone --filter=tree:0 --reference /build/gitref https://github.com/${{ github.repository }} .
+          git fetch origin ${{ github.event.pull_request.head.sha }}
+          git reset --hard ${{ github.event.pull_request.head.sha }}
+          git log -1
+      - name: Build kernel
+        continue-on-error: true
+        run: |
+          mkdir -p /build/logs/
+          cd /build/my-linux && bash .github/scripts/build_ubuntu_defconfig.sh | tee -i /build/logs/all.log
+      - name: Collect build log
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-logs
+          path: /build/logs/*
+      - name: Publish kernel
+        uses: actions/upload-artifact@v4
+        # TODO create modules directory and kernel separately
+        with:
+          name: test-kernel
+          path: /build/rv64__testsuites__plain__gcc/*
+      - name: Publish CI files
+        uses: actions/upload-artifact@v4
+        with:
+          include-hidden-files: true
+          name: ci-files
+          path: /build/my-linux/.github/*
+
+  run-ltp:
+    needs: build-kernel
+    if:  ${{ endsWith(github.head_ref, '_test') }}
+    runs-on: self-hosted
+    timeout-minutes: 50400 # 35 days
+    container:
+      image: ghcr.io/linux-riscv/linaro-tuxrun-dispatcher-riscv64:latest
+      volumes:
+        - /home/github/ccache:/build/ccache
+        - /home/github/gitref:/build/gitref
+        - /tmp:/tmp
+        - /boot:/boot
+        - /lib/modules:/lib/modules
```

Comment:
Why do you need /boot and /lib/modules from the host?

Reply:
I'm testing it right now, don't know if it's a leftover from previous trials or needed by tuxrun.

Reply:
It was a leftover for when I tried using tuxrun with the docker runtime, I removed it.